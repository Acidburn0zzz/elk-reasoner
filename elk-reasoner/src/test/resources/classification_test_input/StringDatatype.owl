Prefix(:=<http://www.semanticweb.org/Test.owl#>)
Prefix(owl:=<http://www.w3.org/2002/07/owl#>)
Prefix(rdf:=<http://www.w3.org/1999/02/22-rdf-syntax-ns#>)
Prefix(xml:=<http://www.w3.org/XML/1998/namespace>)
Prefix(xsd:=<http://www.w3.org/2001/XMLSchema#>)
Prefix(rdfs:=<http://www.w3.org/2000/01/rdf-schema#>)

Ontology(<http://www.semanticweb.org/Test.owl>

EquivalentClasses(:_valid_string_A	DataHasValue(:dp "This is a string!"))
EquivalentClasses(:_valid_string_B	DataHasValue(:dp "Édition française."))
EquivalentClasses(:_valid_string_C	DataHasValue(:dp "AT&amp;T"))
EquivalentClasses(:_valid_string_D	DataHasValue(:dp "12.5"))
EquivalentClasses(:_valid_string_E	DataHasValue(:dp ""))
EquivalentClasses(:_valid_string_F	DataHasValue(:dp "   Отступ 3 пробела   "))
EquivalentClasses(:_valid_string_G	DataHasValue(:dp "Multiline 
									text"))

EquivalentClasses(:AllStrings		DataSomeValuesFrom(:dp xsd:string))
EquivalentClasses(:AnyNonEmptyString	DataSomeValuesFrom(:dp DatatypeRestriction(xsd:string xsd:minLength "1"^^xsd:integer)))
EquivalentClasses(:AnyShortString	DataSomeValuesFrom(:dp DatatypeRestriction(xsd:string xsd:minLength "0"^^xsd:integer xsd:maxLength "5"^^xsd:integer)))
EquivalentClasses(:NumericString	DataSomeValuesFrom(:dp DatatypeRestriction(xsd:string xsd:pattern "[0-9., ]+")))

EquivalentClasses(:InvalidStringA	DataSomeValuesFrom(:dp DatatypeRestriction(xsd:string xsd:maxLength "5"^^xsd:integer xsd:minLength "10"^^xsd:integer)))
EquivalentClasses(:InvalidStringB	DataSomeValuesFrom(:dp DatatypeRestriction(xsd:string xsd:maxLength "-2"^^xsd:integer)))
EquivalentClasses(:InvalidStringC	DataSomeValuesFrom(:dp DatatypeRestriction(xsd:string xsd:minLength "-10"^^xsd:integer)))
)