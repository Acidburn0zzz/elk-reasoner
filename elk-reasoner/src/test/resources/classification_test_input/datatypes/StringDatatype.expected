Prefix(xsd:=<http://www.w3.org/2001/XMLSchema#>)
Prefix(owl:=<http://www.w3.org/2002/07/owl#>)
Prefix(:=<http://www.semanticweb.org/Test.owl#>)
Prefix(xml:=<http://www.w3.org/XML/1998/namespace>)
Prefix(rdf:=<http://www.w3.org/1999/02/22-rdf-syntax-ns#>)
Prefix(rdfs:=<http://www.w3.org/2000/01/rdf-schema#>)

Ontology(<http://www.semanticweb.org/Test.owl>

SubClassOf(:NumericString	:AnyNonEmptyString)
SubClassOf(:AnyNonEmptyString	:AllStrings)
SubClassOf(:AnyShortString	:AllStrings)

SubClassOf(:_valid_string_A	:AnyNonEmptyString)
SubClassOf(:_valid_string_B	:AnyNonEmptyString)
SubClassOf(:_valid_string_C	:AnyNonEmptyString)
SubClassOf(:_valid_string_D	:AnyShortString)
SubClassOf(:_valid_string_D	:NumericString)
SubClassOf(:_valid_string_E	:AnyShortString)
SubClassOf(:_valid_string_F	:AnyNonEmptyString)
SubClassOf(:_valid_string_G	:AnyNonEmptyString)

EquivalentClasses(:InvalidStringA :InvalidStringB :InvalidStringC owl:Nothing)
)
